### DISCUSSION TITLE
How can you add a new sorting function, similar to "sortByTag", but takes five parameters to the alignment track?

### DISCUSSION THREAD

**Discussion by Lilas-w** [GMOD/jbrowse-components#3563] [Q&A] [answered]
https://github.com/Lilas-w/jbrowse-components/tree/dev0.2

I want to add a new sort function to sort by a specific attribute for reads that span the center line, similar to the existing "SortbyTag" function. However, instead of a single "tag", users will submit five parameters (img1). I have already completed the `SortByNEW.tsx` file (img2) and added the label "Sort by NEW..." in `alignments/src/LinearPileupDisplay/model.ts`, same as "Sort By Tag..."(img3).

img1:
![image](https://user-images.githubusercontent.com/81499006/223049523-1c050b59-cce9-4052-8877-ab170721b80b.png)

img2: SortByNEW.tsx
![image](https://user-images.githubusercontent.com/81499006/223047158-6d376115-f0fd-4753-a6b3-6ec91f742c11.png)

img3: 
![image](https://user-images.githubusercontent.com/81499006/223042085-15e2cfff-8966-4126-a886-c23222760f48.png)

 I am unsure about how to both submit mutated data to the model and retrieve data from the model for use in `PileupRenderer/sortUtil.ts`. When I expand the current SortedBy object (as shown in img4 and img5) in `LinearPileupDisplay/model.ts`, the submit button does not work as expected (button code in img6). However, creating a new SortByNew object that imitates SortedBy would significantly impact the current code (as shown in img7).

img4:
![image](https://user-images.githubusercontent.com/81499006/223051714-e21e60ca-d02d-4885-b544-aebc9d29dff9.png)
img5:
![image](https://user-images.githubusercontent.com/81499006/223049983-7c27192e-bc3f-4e71-b125-ba9885622670.png)

img6:
![image](https://user-images.githubusercontent.com/81499006/223050260-01694c0f-ea27-49c4-b398-52d7d334fa29.png)

img7:
![image](https://user-images.githubusercontent.com/81499006/223050848-509abd61-08ba-440c-ae44-068344a2c79a.png)

If you are adding a new sort function that takes five parameters from the users, how will you write code to submit mutated data and retrieve data from the model to use without hurting the current sortByTag function?

**Comment by cmdcolin** [comment, marked as answer]
I checked out your code but it looked like maybe you are on the right track. Adding extra optional fields to the sortedBy object should be fine

If you can narrow down the question I might be able to help further

**Reply by Lilas-w** [reply]
Thank you for letting me know that I'm on the right track. Perhaps I have found the answer.
